### PR TITLE
Add Tinkerbell Cluster Autoscaler E2E Test

### PR DESCRIPTION
**Pull request by jonathanmeier5**
*Issue #, if available:*
https://github.com/aws/eks-anywhere/issues/5883

*Description of changes:*
This PR adds an end to end test verifying that cluster autoscaler works on tinkerbell.

*Testing (if applicable):*
Ran e2e test manually and verified success.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/e2e/tinkerbell_test.go [hunk 1]
```diff
@@ -1932,3 +1932,18 @@ func TestTinkerbellAirgappedKubernetes127UbuntuProxyConfigFlow(t *testing.T) {
 
 	runTinkerbellAirgapConfigProxyFlow(test, "10.80.0.0/16")
 }
+
+func TestTinkerbellKubernetes127UbuntuCuratedPackagesClusterAutoscalerSimpleFlow(t *testing.T) {
+		minNodes := 1
+		maxNodes := 2
```
Comment (ddjjia) [inline test/e2e/tinkerbell_test.go:1938]: can we  run `CheckCuratedPackagesCredentials` there?

File: test/e2e/tinkerbell_test.go [hunk 2]
```diff
@@ -1932,3 +1932,18 @@ func TestTinkerbellAirgappedKubernetes127UbuntuProxyConfigFlow(t *testing.T) {
...
+		test := framework.NewClusterE2ETest(t,
+			framework.NewTinkerbell(t, framework.WithUbuntu127Tinkerbell()),
+			framework.WithClusterFiller(api.WithKubernetesVersion(v1alpha1.Kube127), api.WithWorkerNodeAutoScalingConfig(minNodes, maxNodes)),
+			framework.WithControlPlaneHardware(1),
```
Comment (ddjjia) [inline test/e2e/tinkerbell_test.go:1942]: why need to set cp node number? I just don't see it else where so if it's default to 1?

Comment (jonathanmeier5) [reply test/e2e/tinkerbell_test.go]: I was following other packages examples here. Searching for `WithControlPlaneHardware(1` I do see some other examples, particularly for packages but also for some other types of tests.

I kind of prefer it explicit like this, but if you'd rather I take it out I can do that, just let me know.

**Pull request closed without merging**